[BUG REPORT] Test failures in Fedora

Reading #1123, I realized I had not enabled tests in the Fedora packages of PySCF.
A scratch build on Fedora rawhide is on
https://koji.fedoraproject.org/koji/taskinfo?taskID=129797803

While the tests pass successfully on aarch64, the tests reveal several memory errors on x86_64 and s390x architectures.

On x86_64
```
pyscf/pbc/scf/test/test_rsjk.py .Fatal Python error: Segmentation fault
Current thread 0x00007f887cc10bc0 (most recent call first):
  File "/builddir/build/BUILD/python-pyscf-2.8.0-build/pyscf-2.8.0/pyscf/pbc/scf/rsjk.py", line 388 in _get_jk_sr
  File "/builddir/build/BUILD/python-pyscf-2.8.0-build/pyscf-2.8.0/pyscf/pbc/scf/rsjk.py", line 487 in get_jk
  File "/builddir/build/BUILD/python-pyscf-2.8.0-build/pyscf-2.8.0/pyscf/pbc/scf/test/test_rsjk.py", line 51 in test_get_jk
  File "/usr/lib64/python3.13/unittest/case.py", line 606 in _callTestMethod
  File "/usr/lib64/python3.13/unittest/case.py", line 651 in run
  File "/usr/lib64/python3.13/unittest/case.py", line 707 in __call__
  File "/usr/lib/python3.13/site-packages/_pytest/unittest.py", line 351 in runtest
  File "/usr/lib/python3.13/site-packages/_pytest/runner.py", line 174 in pytest_runtest_call
  File "/usr/lib/python3.13/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/usr/lib/python3.13/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/usr/lib/python3.13/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/usr/lib/python3.13/site-packages/_pytest/runner.py", line 242 in <lambda>
  File "/usr/lib/python3.13/site-packages/_pytest/runner.py", line 341 in from_call
  File "/usr/lib/python3.13/site-packages/_pytest/runner.py", line 241 in call_and_report
  File "/usr/lib/python3.13/site-packages/_pytest/runner.py", line 132 in runtestprotocol
  File "/usr/lib/python3.13/site-packages/_pytest/runner.py", line 113 in pytest_runtest_protocol
  File "/usr/lib/python3.13/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/usr/lib/python3.13/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/usr/lib/python3.13/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/usr/lib/python3.13/site-packages/_pytest/main.py", line 362 in pytest_runtestloop
  File "/usr/lib/python3.13/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/usr/lib/python3.13/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/usr/lib/python3.13/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/usr/lib/python3.13/site-packages/_pytest/main.py", line 337 in _main
  File "/usr/lib/python3.13/site-packages/_pytest/main.py", line 283 in wrap_session
  File "/usr/lib/python3.13/site-packages/_pytest/main.py", line 330 in pytest_cmdline_main
  File "/usr/lib/python3.13/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/usr/lib/python3.13/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/usr/lib/python3.13/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/usr/lib/python3.13/site-packages/_pytest/config/__init__.py", line 175 in main
  File "/usr/lib/python3.13/site-packages/_pytest/config/__init__.py", line 201 in console_main
  File "/usr/bin/pytest", line 8 in <module>
Extension modules: numpy._core._multiarray_umath, numpy.linalg._umath_linalg, scipy._lib._ccallback_c, h5py._errors, h5py.defs, h5py._objects, h5py.h5, h5py.utils, h5py.h5t, h5py.h5s, h5py.h5ac, h5py.h5p, h5py.h5r, h5py._proxy, h5py._conv, h5py.h5z, h5py.h5a, h5py.h5d, h5py.h5ds, h5py.h5g, h5py.h5i, h5py.h5o, h5py.h5f, h5py.h5fd, h5py.h5pl, h5py.h5l, h5py._selector, scipy.linalg._fblas, scipy.linalg._flapack, scipy.linalg.cython_lapack, scipy.linalg._cythonized_array_utils, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._mt19937, numpy.random.mtrand, numpy.random._philox, numpy.random._pcg64, numpy.random._sfc64, numpy.random._generator, scipy.linalg._solve_toeplitz, scipy.linalg._decomp_lu_cython, scipy.linalg._matfuncs_sqrtm_triu, scipy.linalg.cython_blas, scipy.linalg._matfuncs_expm, scipy.linalg._decomp_update, scipy.sparse._sparsetools, _csparsetools, scipy.sparse._csparsetools, scipy.sparse.linalg._dsolve._superlu, scipy.sparse.linalg._eigen.arpack._arpack, scipy.sparse.linalg._propack._spropack, scipy.sparse.linalg._propack._dpropack, scipy.sparse.linalg._propack._cpropack, scipy.sparse.linalg._propack._zpropack, scipy.sparse.csgraph._tools, scipy.sparse.csgraph._shortest_path, scipy.sparse.csgraph._traversal, scipy.sparse.csgraph._min_spanning_tree, scipy.sparse.csgraph._flow, scipy.sparse.csgraph._matching, scipy.sparse.csgraph._reordering, scipy.special._ufuncs_cxx, scipy.special._ufuncs, scipy.special._specfun, scipy.special._comb, scipy.special._ellip_harm_2, scipy.optimize._group_columns, scipy._lib.messagestream, scipy.optimize._trlib._trlib, scipy.optimize._lbfgsb, _moduleTNC, scipy.optimize._moduleTNC, scipy.optimize._cobyla, scipy.optimize._slsqp, scipy.optimize._minpack, scipy.optimize._lsq.givens_elimination, scipy.optimize._zeros, scipy.optimize._highs.cython.src._highs_wrapper, scipy.optimize._highs._highs_wrapper, scipy.optimize._highs.cython.src._highs_constants, scipy.optimize._highs._highs_constants, scipy.linalg._interpolative, scipy.optimize._bglu_dense, scipy.optimize._lsap, scipy.spatial._ckdtree, scipy.spatial._qhull, scipy.spatial._voronoi, scipy.spatial._distance_wrap, scipy.spatial._hausdorff, scipy.spatial.transform._rotation, scipy.optimize._direct, scipy._lib._uarray._uarray (total: 93)
.github/workflows/run_tests.sh: line 19: 14028 Segmentation fault      (core dumped) pytest pyscf/ -s -c pytest.ini pyscf
There are 0 leftover temporary files
error: Bad exit status from /var/tmp/rpm-tmp.otiF2t (%check)
RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.otiF2t (%check)
Child return code was: 1
EXCEPTION: [Error('Command failed: \n # /usr/bin/systemd-nspawn -q -M a827e931b8bc4f4ebb124bccac3f0578 -D /var/lib/mock/f43-build-57618765-6555954/root -a -u mockbuild --capability=cap_ipc_lock --bind=/tmp/mock-resolv.ltytej1j:/etc/resolv.conf --bind=/dev/btrfs-control --bind=/dev/mapper/control --bind=/dev/fuse --bind=/dev/loop-control --bind=/dev/loop0 --bind=/dev/loop1 --bind=/dev/loop2 --bind=/dev/loop3 --bind=/dev/loop4 --bind=/dev/loop5 --bind=/dev/loop6 --bind=/dev/loop7 --bind=/dev/loop8 --bind=/dev/loop9 --bind=/dev/loop10 --bind=/dev/loop11 --console=pipe --setenv=TERM=vt100 --setenv=SHELL=/bin/bash --setenv=HOME=/builddir --setenv=HOSTNAME=mock --setenv=PATH=/usr/bin:/bin:/usr/sbin:/sbin \'--setenv=PROMPT_COMMAND=printf "\\033]0;<mock-chroot>\\007"\' \'--setenv=PS1=<mock-chroot> \\s-\\v\\$ \' --setenv=LANG=C.UTF-8 --resolv-conf=off bash --login -c \'/usr/bin/rpmbuild -bb --noclean --target x86_64 --nodeps /builddir/build/SPECS/python-pyscf.spec\'\n', 1)]
Traceback (most recent call last):
  File "/usr/lib/python3.13/site-packages/mockbuild/trace_decorator.py", line 93, in trace
    result = func(*args, **kw)
  File "/usr/lib/python3.13/site-packages/mockbuild/util.py", line 610, in do_with_status
    raise exception.Error("Command failed: \n # %s\n%s" % (cmd_pretty(command, env), output), child.returncode)
mockbuild.exception.Error: Command failed: 
 # /usr/bin/systemd-nspawn -q -M a827e931b8bc4f4ebb124bccac3f0578 -D /var/lib/mock/f43-build-57618765-6555954/root -a -u mockbuild --capability=cap_ipc_lock --bind=/tmp/mock-resolv.ltytej1j:/etc/resolv.conf --bind=/dev/btrfs-control --bind=/dev/mapper/control --bind=/dev/fuse --bind=/dev/loop-control --bind=/dev/loop0 --bind=/dev/loop1 --bind=/dev/loop2 --bind=/dev/loop3 --bind=/dev/loop4 --bind=/dev/loop5 --bind=/dev/loop6 --bind=/dev/loop7 --bind=/dev/loop8 --bind=/dev/loop9 --bind=/dev/loop10 --bind=/dev/loop11 --console=pipe --setenv=TERM=vt100 --setenv=SHELL=/bin/bash --setenv=HOME=/builddir --setenv=HOSTNAME=mock --setenv=PATH=/usr/bin:/bin:/usr/sbin:/sbin '--setenv=PROMPT_COMMAND=printf "\033]0;<mock-chroot>\007"' '--setenv=PS1=<mock-chroot> \s-\v\$ ' --setenv=LANG=C.UTF-8 --resolv-conf=off bash --login -c '/usr/bin/rpmbuild -bb --noclean --target x86_64 --nodeps /builddir/build/SPECS/python-pyscf.spec'
```

and on s390x
```
*** stack smashing detected ***: terminated
Fatal Python error: Aborted
Current thread 0x000003ff83aef760 (most recent call first):
  File "/builddir/build/BUILD/python-pyscf-2.8.0-build/pyscf-2.8.0/pyscf/scf/_vhf.py", line 782 in rdirect_bindm
  File "/builddir/build/BUILD/python-pyscf-2.8.0-build/pyscf-2.8.0/pyscf/scf/dhf.py", line 938 in _call_veff_ssll
  File "/builddir/build/BUILD/python-pyscf-2.8.0-build/pyscf-2.8.0/pyscf/scf/dhf.py", line 177 in get_jk_coulomb
  File "/builddir/build/BUILD/python-pyscf-2.8.0-build/pyscf-2.8.0/pyscf/scf/dhf.py", line 663 in get_jk
  File "/builddir/build/BUILD/python-pyscf-2.8.0-build/pyscf-2.8.0/pyscf/scf/test/test_dhf.py", line 186 in test_get_jk
  File "/usr/lib64/python3.13/unittest/case.py", line 606 in _callTestMethod
  File "/usr/lib64/python3.13/unittest/case.py", line 651 in run
  File "/usr/lib64/python3.13/unittest/case.py", line 707 in __call__
  File "/usr/lib/python3.13/site-packages/_pytest/unittest.py", line 351 in runtest
  File "/usr/lib/python3.13/site-packages/_pytest/runner.py", line 174 in pytest_runtest_call
  File "/usr/lib/python3.13/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/usr/lib/python3.13/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/usr/lib/python3.13/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/usr/lib/python3.13/site-packages/_pytest/runner.py", line 242 in <lambda>
  File "/usr/lib/python3.13/site-packages/_pytest/runner.py", line 341 in from_call
  File "/usr/lib/python3.13/site-packages/_pytest/runner.py", line 241 in call_and_report
  File "/usr/lib/python3.13/site-packages/_pytest/runner.py", line 132 in runtestprotocol
  File "/usr/lib/python3.13/site-packages/_pytest/runner.py", line 113 in pytest_runtest_protocol
  File "/usr/lib/python3.13/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/usr/lib/python3.13/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/usr/lib/python3.13/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/usr/lib/python3.13/site-packages/_pytest/main.py", line 362 in pytest_runtestloop
  File "/usr/lib/python3.13/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/usr/lib/python3.13/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/usr/lib/python3.13/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/usr/lib/python3.13/site-packages/_pytest/main.py", line 337 in _main
  File "/usr/lib/python3.13/site-packages/_pytest/main.py", line 283 in wrap_session
  File "/usr/lib/python3.13/site-packages/_pytest/main.py", line 330 in pytest_cmdline_main
  File "/usr/lib/python3.13/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/usr/lib/python3.13/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/usr/lib/python3.13/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/usr/lib/python3.13/site-packages/_pytest/config/__init__.py", line 175 in main
  File "/usr/lib/python3.13/site-packages/_pytest/config/__init__.py", line 201 in console_main
  File "/usr/bin/pytest", line 8 in <module>
Extension modules: numpy._core._multiarray_umath, numpy.linalg._umath_linalg, scipy._lib._ccallback_c, h5py._errors, h5py.defs, h5py._objects, h5py.h5, h5py.utils, h5py.h5t, h5py.h5s, h5py.h5ac, h5py.h5p, h5py.h5r, h5py._proxy, h5py._conv, h5py.h5z, h5py.h5a, h5py.h5d, h5py.h5ds, h5py.h5g, h5py.h5i, h5py.h5o, h5py.h5f, h5py.h5fd, h5py.h5pl, h5py.h5l, h5py._selector, scipy.linalg._fblas, scipy.linalg._flapack, scipy.linalg.cython_lapack, scipy.linalg._cythonized_array_utils, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._mt19937, numpy.random.mtrand, numpy.random._philox, numpy.random._pcg64, numpy.random._sfc64, numpy.random._generator, scipy.linalg._solve_toeplitz, scipy.linalg._decomp_lu_cython, scipy.linalg._matfuncs_sqrtm_triu, scipy.linalg.cython_blas, scipy.linalg._matfuncs_expm, scipy.linalg._decomp_update, scipy.sparse._sparsetools, _csparsetools, scipy.sparse._csparsetools, scipy.sparse.linalg._dsolve._superlu, scipy.sparse.linalg._eigen.arpack._arpack, scipy.sparse.linalg._propack._spropack, scipy.sparse.linalg._propack._dpropack, scipy.sparse.linalg._propack._cpropack, scipy.sparse.linalg._propack._zpropack, scipy.sparse.csgraph._tools, scipy.sparse.csgraph._shortest_path, scipy.sparse.csgraph._traversal, scipy.sparse.csgraph._min_spanning_tree, scipy.sparse.csgraph._flow, scipy.sparse.csgraph._matching, scipy.sparse.csgraph._reordering, scipy.special._ufuncs_cxx, scipy.special._ufuncs, scipy.special._specfun, scipy.special._comb, scipy.special._ellip_harm_2, scipy.optimize._group_columns, scipy._lib.messagestream, scipy.optimize._trlib._trlib, scipy.optimize._lbfgsb, _moduleTNC, scipy.optimize._moduleTNC, scipy.optimize._cobyla, scipy.optimize._slsqp, scipy.optimize._minpack, scipy.optimize._lsq.givens_elimination, scipy.optimize._zeros, scipy.optimize._highs.cython.src._highs_wrapper, scipy.optimize._highs._highs_wrapper, scipy.optimize._highs.cython.src._highs_constants, scipy.optimize._highs._highs_constants, scipy.linalg._interpolative, scipy.optimize._bglu_dense, scipy.optimize._lsap, scipy.spatial._ckdtree, scipy.spatial._qhull, scipy.spatial._voronoi, scipy.spatial._distance_wrap, scipy.spatial._hausdorff, scipy.spatial.transform._rotation, scipy.optimize._direct, scipy._lib._uarray._uarray (total: 93)
.github/workflows/run_tests.sh: line 19: 13830 Aborted                 (core dumped) pytest pyscf/ -s -c pytest.ini pyscf
There are 0 leftover temporary files
error: Bad exit status from /var/tmp/rpm-tmp.UP1BMz (%check)
    Bad exit status from /var/tmp/rpm-tmp.UP1BMz (%check)
RPM build errors:
Child return code was: 1
EXCEPTION: [Error('Command failed: \n # /usr/bin/systemd-nspawn -q -M 5c1aa189892b4ab79291c8645fddef6d -D /var/lib/mock/f43-build-57618766-6555954/root -a -u mockbuild --capability=cap_ipc_lock --bind=/tmp/mock-resolv.r6hchrjd:/etc/resolv.conf --bind=/dev/btrfs-control --bind=/dev/mapper/control --bind=/dev/fuse --bind=/dev/loop-control --bind=/dev/loop0 --bind=/dev/loop1 --bind=/dev/loop2 --bind=/dev/loop3 --bind=/dev/loop4 --bind=/dev/loop5 --bind=/dev/loop6 --bind=/dev/loop7 --bind=/dev/loop8 --bind=/dev/loop9 --bind=/dev/loop10 --bind=/dev/loop11 --console=pipe --setenv=TERM=vt100 --setenv=SHELL=/bin/bash --setenv=HOME=/builddir --setenv=HOSTNAME=mock --setenv=PATH=/usr/bin:/bin:/usr/sbin:/sbin \'--setenv=PROMPT_COMMAND=printf "\\033]0;<mock-chroot>\\007"\' \'--setenv=PS1=<mock-chroot> \\s-\\v\\$ \' --setenv=LANG=C.UTF-8 --resolv-conf=off bash --login -c \'/usr/bin/rpmbuild -bb --noclean --target s390x --nodeps /builddir/build/SPECS/python-pyscf.spec\'\n', 1)]
Traceback (most recent call last):
  File "/usr/lib/python3.13/site-packages/mockbuild/trace_decorator.py", line 93, in trace
    result = func(*args, **kw)
  File "/usr/lib/python3.13/site-packages/mockbuild/util.py", line 610, in do_with_status
    raise exception.Error("Command failed: \n # %s\n%s" % (cmd_pretty(command, env), output), child.returncode)
mockbuild.exception.Error: Command failed: 
 # /usr/bin/systemd-nspawn -q -M 5c1aa189892b4ab79291c8645fddef6d -D /var/lib/mock/f43-build-57618766-6555954/root -a -u mockbuild --capability=cap_ipc_lock --bind=/tmp/mock-resolv.r6hchrjd:/etc/resolv.conf --bind=/dev/btrfs-control --bind=/dev/mapper/control --bind=/dev/fuse --bind=/dev/loop-control --bind=/dev/loop0 --bind=/dev/loop1 --bind=/dev/loop2 --bind=/dev/loop3 --bind=/dev/loop4 --bind=/dev/loop5 --bind=/dev/loop6 --bind=/dev/loop7 --bind=/dev/loop8 --bind=/dev/loop9 --bind=/dev/loop10 --bind=/dev/loop11 --console=pipe --setenv=TERM=vt100 --setenv=SHELL=/bin/bash --setenv=HOME=/builddir --setenv=HOSTNAME=mock --setenv=PATH=/usr/bin:/bin:/usr/sbin:/sbin '--setenv=PROMPT_COMMAND=printf "\033]0;<mock-chroot>\007"' '--setenv=PS1=<mock-chroot> \s-\v\$ ' --setenv=LANG=C.UTF-8 --resolv-conf=off bash --login -c '/usr/bin/rpmbuild -bb --noclean --target s390x --nodeps /builddir/build/SPECS/python-pyscf.spec'
```

In some of the [build logs](https://kojipkgs.fedoraproject.org//packages/python-pyscf/2.8.0/2.fc42/data/logs/x86_64/build.log), there are messages like
```
In file included from /builddir/build/BUILD/python-pyscf-2.8.0-build/pyscf-2.8.0/pyscf/lib/dft/nr_numint.c:23:
/builddir/build/BUILD/python-pyscf-2.8.0-build/pyscf-2.8.0/pyscf/lib/gto/grid_ao_drv.h:30:9: warning: ‘SIMDD’ redefined
   30 | #define SIMDD           8
      |         ^~~~~
In file included from /builddir/build/BUILD/python-pyscf-2.8.0-build/pyscf-2.8.0/pyscf/lib/gto/grid_ao_drv.h:23:
/usr/include/cint.h:156:9: note: this is the location of the previous definition
  156 | #define SIMDD   2
```

This could be related to the testing failure.

--- a/pyscf/lib/pbc/cint2e.c
+++ b/pyscf/lib/pbc/cint2e.c
@@ -27,6 +27,12 @@
 #include "np_helper/np_helper.h"
 #include "pbc/pbc.h"
 
+#ifdef QCINT_VERSION
+int compiled_with_qcint = 1;
+#else
+int compiled_with_qcint = 0;
+#endif
+
 void CINTinit_int2e_EnvVars(CINTEnvVars *envs, int *ng, int *shls,
                             int *atm, int natm, int *bas, int nbas, double *env);
 extern void CINTgout2e();
@@ -167,7 +173,12 @@ int PBCint2e_loop(double *gctr, int *cell0_shls, int *bvk_cells, int cutoff,
         if (envs_cint->opt == NULL) {
                 intor_loop = &CINT2e_loop_nopt;
         } else if (x_ctr[0] == 1 && x_ctr[1] == 1 && x_ctr[2] == 1 && x_ctr[3] == 1) {
+#ifdef QCINT_VERSION
+                // In qcint, CINT2e_1111_loop is problematic can cause seg fault.
+                intor_loop = &CINT2e_loop;
+#else
                 intor_loop = &CINT2e_1111_loop;
+#endif
         } else {
                 intor_loop = &CINT2e_loop;
         }

--- a/pyscf/lib/pbc/nr_direct.c
+++ b/pyscf/lib/pbc/nr_direct.c
@@ -568,6 +568,7 @@ static void approx_bvk_rcond0(float *rcond, int ish0, int ish1, BVKEnvs *envs_bv
                 for (jseg = jseg0; jseg < jseg1; jseg++) {
                         jsh0 = seg2sh[jseg];
                         jsh1 = seg2sh[jseg+1];
+                        if (jsh0 >= jsh1) continue;
                         aj = env[bas(PTR_EXP, jsh0) + bas(NPRIM_OF, jsh0) - 1];
                         aij = ai + aj;
                         ci = ai / aij;

--- a/pyscf/pbc/df/incore.py
+++ b/pyscf/pbc/df/incore.py
@@ -38,6 +38,17 @@ LOG_ADJUST = 32
 
 libpbc = lib.load_library('libpbc')
 
+def verify_cint_backend():
+    compiled_with_qcint = ctypes.c_int.in_dll(libpbc, 'compiled_with_qcint').value
+    if compiled_with_qcint:
+        if not hasattr(libpbc, 'CINTgout2e_simd1'):
+            raise RuntimeError(
+                'PySCF was compiled with qcint, but the loaded integral library is cint')
+    else:
+        if hasattr(libpbc, 'CINTgout2e_simd1'):
+            raise RuntimeError(
+                'PySCF was compiled with cint, but the loaded integral library is qcint')
+
 def make_auxcell(cell, auxbasis=None):
     '''
     See pyscf.df.addons.make_auxmol
@@ -321,6 +332,7 @@ class Int3cBuilder(lib.StreamObject):
             reindex_k = np.asarray(reindex_k, dtype=np.int32)
             nkpts_ij = reindex_k.size
 
+        verify_cint_backend()
         drv = libpbc.PBCfill_nr3c_drv
 
         # is_pbcintor controls whether to use memory efficient functions

--- a/pyscf/pbc/scf/rsjk.py
+++ b/pyscf/pbc/scf/rsjk.py
@@ -39,7 +39,7 @@ from pyscf.pbc.df import aft, rsdf_builder, aft_jk
 from pyscf.pbc.df import ft_ao
 from pyscf.pbc.df.df_jk import (zdotNN, zdotCN, zdotNC, _ewald_exxdiv_for_G0,
                                 _format_dms, _format_kpts_band, _format_jks)
-from pyscf.pbc.df.incore import libpbc, _get_cache_size, LOG_ADJUST
+from pyscf.pbc.df.incore import libpbc, _get_cache_size, LOG_ADJUST, verify_cint_backend
 from pyscf.pbc.lib.kpts_helper import (is_zero, kk_adapted_iter,
                                        group_by_conj_pairs, intersection)
 from pyscf import __config__
@@ -207,7 +207,9 @@ class RangeSeparatedJKBuilder(lib.StreamObject):
                 log.debug('supmol_d nbas = %d cGTO = %d', supmol_d.nbas, supmol_d.nao)
         log.timer_debug1('initializing supmol', *cpu0)
 
-        self._cintopt = lib.c_null_ptr()
+        verify_cint_backend()
+        self._cintopt = _vhf.make_cintopt(supmol_sr._atm, supmol_sr._bas,
+                                          supmol_sr._env, 'int2e')
         nbas = supmol_sr.nbas
         qindex = np.empty((3,nbas,nbas), dtype=np.int16)
         ao_loc = supmol_sr.ao_loc

